python/tk_multi_publish2/publish_tree_widget/publish_tree_widget.py

```diff
@@ -165,6 +165,13 @@ def _build_item_tree_r(self, item, checked, level, tree_parent):
         if not item.checked:
             ui_item.set_check_state(QtCore.Qt.Unchecked)
 
+        # If we have a bunch of active tasks followed by a bunch of inactive tasks, the addition
+        # of the inactive tasks does not trigger an update of the parent's checkbox (because the update
+        # relies on the checkbox.state_changed and the default is unchecked, so inactive tasks do not
+        # trigger a state change). Force a recalculation to keep the parent's check_state
+        # correct in all situation
+        ui_item.recompute_check_state()
```

Comment:
This may be fine, but I just wanted to check:
on line 166 just above, it sets the state to unchecked if the item is not checked.
If this item is unchecked, do actually want to recompute the check state based on it's children, since the item should always be unchecked, regardless of the children's state? Or is my understanding wrong?

Reply:
If you have an item that is unchecked but has tasks that are checked, before you get to the line that I added, the tasks are added to the item which, by triggering the state_change signal, also update the parent. So the parent is now checked. Then there's a check to see if the parent item should be inactive, and if so, it's set to unchecked. At this point, since the parent was previously checked, we get another signal telling all the children to uncheck themselves. My recompute addition should do nothing in this case because everything is consistent. 

I didn't put it in an else statement because there's another case that this needs to handle. If the parent item is set to active but it has all tasks that are inactive. This works fine in UI, but if you just use the underlying code, adding the children doesn't update the parent status since they're all inactive. That means you end up with a parent that is checked and all its items are unchecked. 

Now that I think about it, I should probably add this as a test :)